src/Serialization.js

```diff
@@ -12,13 +12,17 @@ const setter = prop => obj => value => () => obj["set_" + prop](value);
 exports.hashTransaction = body => () => lib.hash_transaction(body);
 
 exports.newBigNum = maybe => string => {
-  try {
-    return maybe.just(lib.BigNum.from_str(string));
-  } catch (_) {
-    return maybe.nothing;
+  // this is needed because try/catch overuse breaks runtime badly
+  // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
+  if (/^-?[0-9]+$/.test(str)) {
+      return maybe.just(lib.BigInt.from_str(str));
```

Comment:
We should not remove try/catch entirely. We want to just add more checks _before_ using try-catch. The problem with wasm-bindgen arises only when an object is allocated in a _failing_ try block.